mula/scheduler/alembic/versions/0008_restructure.py

```diff
@@ -0,0 +1,96 @@
+"""restructure
+
+Revision ID: 0008
+Revises: 0007
+Create Date: 2024-04-01 15:45:35.429784
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+import scheduler
+
+# revision identifiers, used by Alembic.
+revision = "0008"
+down_revision = "0007"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "schemas",
+        sa.Column("id", scheduler.utils.datastore.GUID(), nullable=False),
+        sa.Column("scheduler_id", sa.String(), nullable=False),
+        sa.Column("hash", sa.String(length=32), nullable=True),
+        sa.Column("data", postgresql.JSONB(astext_type=sa.Text()), nullable=False),
+        sa.Column("enabled", sa.Boolean(), nullable=False),
+        sa.Column("schedule", sa.String(), nullable=True),
+        sa.Column("deadline_at", sa.DateTime(timezone=True), nullable=True),
+        sa.Column("created_at", sa.DateTime(timezone=True), server_default=sa.text("now()"), nullable=False),
+        sa.Column("modified_at", sa.DateTime(timezone=True), server_default=sa.text("now()"), nullable=False),
+        sa.PrimaryKeyConstraint("id"),
+    )
+    op.drop_index("ix_items_hash", table_name="items")
+    op.drop_table("items")
+    op.add_column("tasks", sa.Column("schema_id", scheduler.utils.datastore.GUID(), nullable=True))
+    op.add_column("tasks", sa.Column("hash", sa.String(length=32), nullable=True))
+    op.add_column("tasks", sa.Column("priority", sa.Integer(), nullable=True))
+    op.add_column("tasks", sa.Column("data", postgresql.JSONB(astext_type=sa.Text()), nullable=False))
+    op.alter_column("tasks", "scheduler_id", existing_type=sa.VARCHAR(), nullable=False)
+    op.drop_index("ix_tasks_p_item_hash", table_name="tasks")
+    op.create_index(op.f("ix_tasks_hash"), "tasks", ["hash"], unique=False)
+    op.create_foreign_key(None, "tasks", "schemas", ["schema_id"], ["id"], ondelete="SET NULL")
+    op.drop_column("tasks", "p_item")
```

Comment:
Shouldn't we be more careful and actually migrate this data to the new data field? Or does that contain a new kind of data structure?

Reply:
Yeah it going to change to data, the prioritized wrapper is going to be removed. However, you make a good point the `data` in the `p_item` should be migrated to the new `data` field

Reply:
@Donnype mentioned he has done this work in bytes as well